client/my-sites/domains/transfer-domain/index.jsx

```diff
@@ -69,14 +70,19 @@ export class TransferDomain extends Component {
 	addDomainToCart = ( suggestion ) => {
 		const { selectedSiteSlug } = this.props;
 
-		addItem(
-			domainRegistration( {
-				productSlug: suggestion.product_slug,
-				domain: suggestion.domain_name,
-			} )
-		);
-
-		page( '/checkout/' + selectedSiteSlug );
+		this.props.shoppingCartManager
+			.addProductsToCart( [
+				fillInSingleCartItemAttributes(
+					domainRegistration( {
+						productSlug: suggestion.product_slug,
+						domain: suggestion.domain_name,
+					} ),
+					this.props.productsList
+				),
+			] )
+			.then( () => {
+				page( '/checkout/' + selectedSiteSlug );
+			} );
```

Comment:
Not mandatory but we have already destructured `selectedSiteSlug`, we coud do the same for `shoppingCartManager` and `productsList`.